src/api/fetch.js

```diff
@@ -1,24 +1,28 @@
 import baseURL from "./base_url";
+import { Scope } from "./scopes";
 const fetchRetry = require("fetch-retry")(fetch, {
   retries: 0,
   retryDelay: function (attempt) {
     return Math.pow(2, attempt) * 15000;
   },
 });
 
-export async function* indexGenerator(path, auth) {
+export async function* indexGenerator(path, auth, scope = new Scope()) {
```

Comment:
From an API design standpoint, the optionality of this parameter should be clear in the `index` methods, not here.

Reply:
I've also made it clear in the index action where applicable. We also need the optionality here, since not all index actions will provide a scope